src/riak_kv_qry_compiler.erl

```diff
@@ -97,16 +97,22 @@ run_select(Select, Row, InitialState) ->
 
 %% @priv
 run_select2([], _, _, Acc) ->
```

Comment:
Sorry, I missed that note in time.

I didn't have measurements at the time, but we are talking about a full list reverse times two vs an append of the whole list.
I recon that append and reverse have the same running time, so my version could be up to 50% faster.

Today I did an experiment using the code below and depending on the length of the list of lists and the lengths of the individual lists my version with body recursion is between 30-60% faster than the tail recursive version.

```erlang
-module(append_reverse).

-compile(export_all).

list_of_lists() ->
    N =  3 + rand:uniform(6),
    [ random_list() || _ <- lists:seq(1,N) ].

random_list() ->
    Len = rand:uniform(5) - 1,
    lists:seq(1,Len).

my_append([]) ->
    [];
my_append([X|Xs]) ->
    prepend(X, Xs).

prepend([], Xs) ->
    [ [] | my_append(Xs) ];
prepend(Ys, Xs) ->
    Ys ++ my_append(Xs).

my_fold(Xs) ->
    my_fold2(Xs, []).

my_fold2([], Acc) ->
    lists:reverse(Acc);
my_fold2([X|Xs], Acc) ->
    Acc2 = prepend2(X, Acc),
    my_fold2(Xs, Acc2).

prepend2([_|_]=Xs, Acc) ->
    lists:reverse(Xs) ++ Acc;
prepend2(_, Acc) ->
    [ [] | Acc ].


test1() ->
    Data = list_of_lists(),
    {Time1, Res1} = timer:tc( fun my_append/1, [Data]),
    {Time2, Res2} = timer:tc( fun my_fold/1, [Data]),
    Res1 = Res2,
    {Time1, Time2}.

test() ->
    Samples = [ test1() || _ <- lists:seq(1,10000) ],
    {AppendTimes, FoldTimes} = lists:unzip(Samples),
    {stats(AppendTimes), stats(FoldTimes)}.

stats(Samples) ->
    Min = lists:min(Samples),
    Max = lists:max(Samples),
    Avg = lists:sum(Samples) / length(Samples),
    {Min, Avg, Max}.
```